src/test/java/org/folio/innreach/controller/InnReachTransactionControllerTest.java

```diff
@@ -688,6 +701,128 @@ void return200HttpCode_and_sendRequest_whenItemHoldTransactionCreated() {
   }
 
+  @Test
+  @Sql(scripts = {
+    "classpath:db/central-server/pre-populate-central-server.sql",
+    "classpath:db/mtype-mapping/pre-populate-material-type-mapping.sql",
+    "classpath:db/central-patron-type-mapping/pre-populate-central-patron_type-mapping-table.sql"
+  })
+  void return200HttpCode_and_sendRequest_whenItemHoldTransactionCreated_chosenPickLocationFromTransaction() {
+    var inventoryItemDTO = mockInventoryClient();
+    inventoryItemDTO.setStatus(IN_TRANSIT);
+    inventoryItemDTO.setTitle(randomAlphanumeric(500));
+    var requestDTO = createRequestDTO();
+    requestDTO.setItemId(inventoryItemDTO.getId());
+    modifyCentralServer(UUID.fromString(PRE_POPULATED_CENTRAL_SERVER_ID));
+    var servicePoint = new ServicePointsClient.ServicePoint();
+    servicePoint.setId(UUID.fromString(PRE_POPULATED_CENTRAL_SERVER_ID));
+
+    when(circulationClient.queryRequestsByItemId(inventoryItemDTO.getId())).thenReturn(ResultList.of(1,
+      List.of(requestDTO)));
+    var user = mockUserClient();
+    var requestPreference = new RequestPreferenceDTO(user.getId(), randomUUID());
+    when(requestPreferenceClient.getUserRequestPreference(user.getId())).thenReturn(ResultList.of(1,
+      List.of(requestPreference)));
+    when(circulationClient.sendRequest(any(RequestDTO.class))).then((Answer<RequestDTO>) invocationOnMock -> {
+      var sentRequest = (RequestDTO) invocationOnMock.getArgument(0);
+      sentRequest.setId(randomUUID());
+      return sentRequest;
+    });
+    when(servicePointsClient.queryServicePointByCode(PRE_POPULATED_PICK_LOCATION_CODE))
+      .thenReturn(ResultList.of(1, List.of(servicePoint)));
+
+    var itemHoldDTO = deserializeFromJsonFile(
+      "/inn-reach-transaction/create-item-hold-request.json", TransactionHoldDTO.class);
+    itemHoldDTO.setCentralPatronType(PRE_POPULATED_CENTRAL_PATRON_TYPE);
+    itemHoldDTO.setItemId(inventoryItemDTO.getHrid());
+
+
+    var responseEntity = testRestTemplate.postForEntity(
+      "/inn-reach/d2ir/circ/itemhold/{trackingId}/{centralCode}", new HttpEntity<>(itemHoldDTO, headers), InnReachResponseDTO.class, TRACKING_ID,
+      PRE_POPULATED_CENTRAL_SERVER_CODE);
+
+    assertEquals(HttpStatus.OK, responseEntity.getStatusCode());
+    assertTrue(responseEntity.hasBody());
+    assertEquals("ok", responseEntity.getBody().getStatus());
+
+    await().atMost(ASYNC_AWAIT_TIMEOUT).untilAsserted(() ->
+      verify(repository, atLeastOnce()).save(
+        argThat((InnReachTransaction t) -> t.getHold().getFolioRequestId() != null)));
+
+    var transaction = repository.fetchOneByTrackingId(TRACKING_ID);
+    var newRequestCaptor = ArgumentCaptor.forClass(RequestDTO.class);
+
+    verify(circulationClient).sendRequest(newRequestCaptor.capture());
+
+    var newRequest = newRequestCaptor.getValue();
+
+    verify(servicePointsClient).queryServicePointByCode(transaction.get().getHold().getPickupLocation().getPickupLocCode());
```

Comment:
you defined a constant for this value - `PRE_POPULATED_PICK_LOCATION_CODE`

Reply:
Fixed